server/utils/placementApplications/review.test.ts

```diff
@@ -0,0 +1,116 @@
+import type { Request } from 'express'
+import { DeepMocked, createMock } from '@golevelup/ts-jest'
+import type { PlacementApplicationDecision } from '@approved-premises/api'
+import { PlacementApplicationReview } from './review'
+
+describe('PlacementApplicationReview', () => {
+  const id = 'some-uuid'
+
+  afterEach(() => {
+    jest.clearAllMocks()
+  })
+
+  it('initialises with expected values', () => {
+    const request: DeepMocked<Request> = createMock<Request>({
+      params: { id },
+    })
+
+    const review = new PlacementApplicationReview(request, 'review')
+
+    expect(review.applicationId).toEqual(id)
+    expect(review.body).toEqual({})
+    expect(review.step).toEqual('review')
+  })
+
+  describe('validate', () => {
+    const request: DeepMocked<Request> = createMock<Request>({
+      params: { id },
+      session: createMock<Request['session']>({
+        placementApplicationDecisions: {},
+      }),
+    })
+
+    it('records no errors when called with correct review args', () => {
+      const body = {
+        summaryOfChanges: 'some changes',
+        decision: null as PlacementApplicationDecision,
+        decisionSummary: '',
+      }
+
+      const review = new PlacementApplicationReview(request, 'review')
+
+      review.validate(body)
+
+      expect(review.errors).toEqual({})
+    })
+
+    it('records an error when called with missing review body', () => {
+      const body = { summaryOfChanges: '', decision: null as PlacementApplicationDecision, decisionSummary: '' }
+
+      const review = new PlacementApplicationReview(request, 'review')
+
+      review.validate(body)
+
+      expect(review.errors).toEqual({ summaryOfChanges: 'You must provide a summary of the changes' })
+    })
+
+    it('records no errors when called with correct decision args', () => {
+      const body = { summaryOfChanges: '', decision: 'accepted' as const, decisionSummary: 'a summary' }
+
+      const review = new PlacementApplicationReview(request, 'decision')
+
+      review.validate(body)
+
+      expect(review.errors).toEqual({})
+    })
+
+    it('records an error when called with missing decision body', () => {
+      const body = { summaryOfChanges: '', decision: null as PlacementApplicationDecision, decisionSummary: '' }
```

Comment:
It might be good to have a default `body` that we use for each of these tests that we mutate for each one. I think it might be easier to see what is changing between each test. Don't forget to reset it to the default in the `afterEach` though.